elasticsearch_dsl/document.py

```diff
@@ -262,7 +265,15 @@ def to_dict(self, include_meta=False):
         meta['_source'] = d
         return meta
 
-    def update(self, using=None, index=None, **fields):
+    def update(self, using=None, index=None, detect_noop=True, doc_as_upsert=False, **fields):
+        """ Update this document in Elasticsearch.
+
+        :param using:
+        :param index:
+        :param detect_noop: check if values have changed to avoid unnecessary write operation. Default: True.
+        :param doc_as_upsert: index document if it doesn't exist. Default: False.
+        :param fields: kwargs of additional fields
```

Comment:
Thanks for the patch, in the rest of the code we use `:arg NAME:` syntax instead of `:param NAME:`, could you please change it to be consistent? Thanks!